examples/2d/texture_atlas.rs

```diff
@@ -219,10 +219,10 @@ fn create_texture_atlas(
             continue;
         };
 
-        texture_atlas_builder.add_texture(Some(id), texture);
+        texture_atlas_builder = texture_atlas_builder.add_texture(Some(id), texture);
```

Comment:
This doesn't seem like a win and just makes it much easier to mess up if you forget to do the assignment and we can't really warn users easily if they were using the previous code so it will break and be hard to debug.

Reply:
I don't think there's anything wrong with using mutation here

Reply:
Probably better to have this method take a mutable reference and return a mutable reference, this is a Pattern quite used for builders IIRC